shell/platform/windows/client_wrapper/flutter_view_controller.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "include/flutter/flutter_view_controller.h"
+
+#include <algorithm>
+#include <iostream>
+
+namespace flutter {
+
+FlutterViewController::FlutterViewController(
+    const std::string& icu_data_path)
+    : icu_data_path_(icu_data_path) {
+}
+
+FlutterViewController::~FlutterViewController() {
+  if (controller_) {
+    FlutterDesktopDestroyView(controller_);
+  }
+}
+
+FlutterView FlutterViewController::CreateFlutterView(
+    int width,
+    int height,
+    const std::string& assets_path,
+    const std::vector<std::string>& arguments) {
+
+  if (controller_) {
+    std::cerr << "Only one Flutter window can exist at a time." << std::endl;
+    return nullptr;
+  }
+
+  std::vector<const char*> engine_arguments;
+  std::transform(
+      arguments.begin(), arguments.end(), std::back_inserter(engine_arguments),
+      [](const std::string& arg) -> const char* { return arg.c_str(); });
+  size_t arg_count = engine_arguments.size();
+
+  controller_ = FlutterDesktopCreateView(
+      width,
+      height,
+      assets_path.c_str(), icu_data_path_.c_str(),
+      arg_count > 0 ? &engine_arguments[0] : nullptr, arg_count);
+  if (!controller_) {
+    std::cerr << "Failed to create window." << std::endl;
+    return nullptr;
+  }
+  view_ =
+      std::make_shared<FlutterViewWin32>(FlutterDesktopGetView(controller_));
```

Comment:
Shared pointers are very strongly discouraged in the C++ style guide. Could you give a high-level overview of why we need shared pointers rather than unique_ptr?

I'm also concerned because it's non-obvious from the local code what happens if this object outlives the destruction of the controller (and thus engine).

Reply:
The change to consolidate view and view controller will fix this as a side effect.